container/lxd/container.go

```diff
@@ -39,15 +41,20 @@ type ContainerSpec struct {
 // Note that we pass these through as supplied. If an instance type constraint
 // has been specified along with specific cores/mem constraints,
 // LXD behaviour is to override with the specific ones even when lower.
-func (c *ContainerSpec) ApplyConstraints(cons constraints.Value) {
+func (c *ContainerSpec) ApplyConstraints(serverVersion string, cons constraints.Value) {
 	if cons.HasInstanceType() {
 		c.InstanceType = *cons.InstanceType
 	}
 	if cons.HasCpuCores() {
 		c.Config["limits.cpu"] = fmt.Sprintf("%d", *cons.CpuCores)
 	}
 	if cons.HasMem() {
-		c.Config["limits.memory"] = fmt.Sprintf("%dMiB", *cons.Mem)
+		// LXD versions < 3 do not recognise the correct "MiB" notation.
+		template := "%dMiB"
+		if v, err := strconv.Atoi(strings.Split(serverVersion, ".")[0]); err == nil && v < 3 {
```

Comment:
Rather than doing this, should we not be doing something like https://github.com/lxc/lxd/blob/master/shared/version/version.go#L51 then we can extract major/minor/patch really easily.